frontend/src/components/bucket/BucketConfigForm.vue

```diff
@@ -0,0 +1,154 @@
+<script setup lang="ts">
+import Button from 'primevue/button';
+import { Form } from 'vee-validate';
+import { object, string } from 'yup';
+import { useBucketStore } from '@/store';
+import { useToast } from 'primevue/usetoast';
+import Password from '@/components/form/Password.vue';
+import TextInput from '@/components/form/TextInput.vue';
+import type { Bucket } from '@/interfaces';
+
+const props = defineProps<{
+  isUpdate: boolean;
+  bucket?: Bucket;
+}>();
+
+const bucketStore = useBucketStore();
+const toast = useToast();
+
+const emit = defineEmits(['submit-bucket-config', 'cancel-bucket-config']);
+
+const initialValues = {
+  bucketName: props.bucket?.bucketName,
+  bucket: props.bucket?.bucket,
+  endpoint: props.bucket?.endpoint,
+  accessKeyId: props.bucket?.accessKeyId,
+  secretAccessKey: props.bucket?.secretAccessKey,
+  key: props.bucket?.key
+};
+
+const schema = object({
+  bucketName: string().max(255).required().label('Bucket name'),
+  bucket: string().max(255).required().label('Bucket'),
+  endpoint: string().max(255).required().label('Endpoint'),
+  accessKeyId: string().max(255).required().label('Access Key ID'),
+  secretAccessKey: string().max(255).required().label('Secret Access Key'),
+  key: string().max(255).required().label('Key'),
+});
+
+const onSubmit = async (values: any) => {
+
+  try {
+    const formBucket = {
+      bucketName: values.bucketName,
+      bucket: values.bucket,
+      endpoint: values.endpoint,
+      accessKeyId: values.accessKeyId !== 'REDACTED' && values.accessKeyId !== '' ?
```

Comment:
having noticed that these two fields can't be empty (due to validation rules).. i guess you can just do:
`accessKeyId: values.accessKeyId !== 'REDACTED' ? values.accessKeyId : undefined`

Reply:
not sure we found the best solution by having 'REDACTED' as initial value.
might have to re-think the best UX